misc/h2olog

```diff
@@ -0,0 +1,128 @@
+#! /bin/sh
+exec ${H2O_PERL:-perl} -x $0 "$@"
+#! perl
+
+use strict;
+use warnings;
+use Getopt::Long qw(:config no_ignore_case bundling);
+use IO::Socket::UNIX;
+use JSON;
+use POSIX qw(strftime);
+use Socket qw(SOCK_STREAM);
+
+my ($unix_socket, $debug, $http_mode, $write_file, @query);
+my $sample_rate = 1;
+
+GetOptions(
+    'u|unix=s'            => \$unix_socket,
+    'A|address=s'         => sub {
+        push @query, "address=" . uri_escape($_[1]);
+    },
+    'a|application-data'  => sub {
+        push @query, "application-data=1";
+    },
+    'd|debug'             => \$debug,
+    'H|http'              => \$http_mode,
+    'N|sni=s'             => sub {
+        push @query, "sni=" . uri_escape($_[1]);
+    },
+    's|response-header=s' => sub {
+        push @query, "response-header=" . uri_escape($_[1]);
+    },
+    'S|sample-rate=f'     => sub {
+        push @query, "sample-rate=" . uri_escape($_[1]);
+    },
+    't|trace=s'           => sub {
+        push @query, "trace=" . uri_escape($_[1]);
+    },
```

Comment:
As can be seen, the command line options are converted to query parameters, and are passed the h2olog handler like: `/.well-known/h2olog?<query-parameters>`.